packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1609,62 +1652,78 @@ actual object RealmInterop {
         realm: RealmPointer,
         classKey: ClassKey,
         query: String,
-        args: RealmQueryArgumentList
+        args: RealmQueryArgumentList,
     ): RealmQueryPointer {
-        return LongPointerWrapper(
-            realmc.realm_query_parse(
-                realm.cptr(),
-                classKey.key,
-                query,
-                args.size,
-                args.head
+        try {
+            return LongPointerWrapper(
+                realmc.realm_query_parse(
+                    realm.cptr(),
+                    classKey.key,
+                    query,
+                    args.size,
+                    args.head
+                )
             )
-        )
+        } finally {
+            realmc.delete_queryArgArray(args.head)
```

Comment:
I don't think we release everything here. The query argument array holds a pointer to a value array allocated in `JvmMemAllocator.allocRealmValueList`. Maybe not showing up in the benchmarks if you are not testing with list query-arguments (`query("x in $0", [1, 2, 3]`)